### PR TITLE
fix(link): if to is an object and to.path exists, use to.path has href fallback

### PR DESCRIPTION
**Pull request by tmorehouse**

### CODE REVIEW

**Comment by alexsasharegan** [comment]
It's this necessary? View router already does this for us I thought.

**Comment by tmorehouse** [comment]
Maybe it does...  so should we even set `href` at all if `to` is provided? (or maybe always set to `'#'` ?

**Comment by tmorehouse** [comment]
This used to be in the original b-link code from way back (before my time).

I think it was a fallback in case vue-router wasn't available

**Comment by alexsasharegan** [comment]
Then maybe it should check for vue-router first. Maybe it could use `Boolean(parent.$router)`?

**Comment by tmorehouse** [comment]
Is there a `$isRouter` prop on the vue instance?

**Comment by tmorehouse** [comment]
Note... just `vm.$router`

**Comment by tmorehouse** [comment]
Actually it appears that router-link ignores the href prop.

**Comment by pi0** [comment]
@tmorehouse We have initially made `b-link` abstraction to make BV more portable not depending on special libraries like vue-router and also have a tight integration with the router if exists. This made many big challenges. First that router-link itself creates href too so having both on our custom component makes many problems specially with SSR renders. But i think with the The `parent.$router` guard in this PR is safe as it just happens when there is **no** router :)

**Comment by tmorehouse** [comment]
Should we event set an `href` (right now defaults to `#` if `to` is provided and there is `parent.$router` defined).  Should it maybe return null or empty string?

**Comment by pi0** [comment]
@tmorehouse, To be honest, can't remind the exact story and reason. But I think it was '#' to show a correct cursor on drop-down, btn etc when router is available but `to` is not specified

**Comment by pi0** [comment]
Ah it's getting more complicated :)) we need a psudo like code to specify b-link's behavior 😆

**Comment by tmorehouse** [comment]
maybe if `$router` is defined and `to` provided, we return `null` for HREF, and let router link generate the HREF. but if `to` is provided and `$router` not avail, return `to/to.path`,

File: lib/components/link.js [hunk 1]
```diff
@@ -89,12 +89,16 @@ function computeTag(props) {
     return props.to && !props.disabled ? "router-link" : "a";
```
Comment (tmorehouse) [inline lib/components/link.js]: Maybe this should check for `parent.$router` as well?

Comment (pi0) [reply lib/components/link.js]: Yes i think it should too.

Comment (tmorehouse) [reply lib/components/link.js]: Just need to make sure its pass `parent` as the next arg

**Comment by tmorehouse** [comment]
Closing s this has been superseded by #1016

**Pull request closed without merging**